src/Workspaces/CSharp/Portable/FindSymbols/CSharpDeclaredSymbolInfoFactoryService.cs

```diff
@@ -147,7 +154,16 @@ private static void ProcessUsings(List<Dictionary<string, string>> aliasMaps, Sy
             }
         }
 
-        public override bool TryGetDeclaredSymbolInfo(StringTable stringTable, SyntaxNode node, string rootNamespace, out DeclaredSymbolInfo declaredSymbolInfo)
+        protected override void AddDeclaredSymbolInfosWorker(
+            SyntaxNode container,
+            MemberDeclarationSyntax node,
+            StringTable stringTable,
+            ArrayBuilder<DeclaredSymbolInfo> declaredSymbolInfos,
+            Dictionary<string, string> aliases,
+            Dictionary<string, ArrayBuilder<int>> extensionMethodInfo,
+            string containerDisplayName,
+            string fullyQualifiedContainerName,
```

Comment:
a core change is that these two strings are now computed by teh caller and passed into all children, saving 5% of allocs in NavTo as reported by @sharwell .